tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -722,6 +745,25 @@ const DeviceCommand EnqueueWriteBufferCommand::assemble_device_command(uint32_t
     command.set_consumer_cb_num_pages(consumer_cb_num_pages);
     command.set_num_pages(num_pages);
 
+    // Targeting fast dispatch on remote device means commands have to be tunneled through ethernet
+    bool route_through_ethernet = not device->is_mmio_capable();
+    if (route_through_ethernet) {
+        uint32_t router_cb_num_pages = get_consumer_data_buffer_size(true) / padded_page_size;
+        if (router_cb_num_pages >= DeviceCommand::SYNC_NUM_PAGES) {
+            router_cb_num_pages = (router_cb_num_pages / DeviceCommand::SYNC_NUM_PAGES) * DeviceCommand::SYNC_NUM_PAGES;
+            command.set_producer_router_transfer_num_pages(router_cb_num_pages / DeviceCommand::SYNC_NUM_PAGES);
+            command.set_consumer_router_transfer_num_pages(router_cb_num_pages / DeviceCommand::SYNC_NUM_PAGES);
+        } else {
+            command.set_producer_router_transfer_num_pages(1);
+            command.set_consumer_router_transfer_num_pages(1);
+        }
```

Comment:
I'm thinking we don't event need this logic, since I believe the cores take the min(num_pages_left, num sync)

Reply:
yeah I can remove this as well as the same condition for producer direct to consumer